com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HighDefinitionRenderPipelineProjectorEditor.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    [CustomEditorForRenderPipeline(typeof(Projector), typeof(HDRenderPipelineAsset))]
+    [CanEditMultipleObjects]
+    public class HighDefinitionRenderPipelineProjectorEditor : Editor
+    {
+        const string k_Message = "The active render pipeline (HDRP) does not support the Projector component. Use the Decal Projector component instead.";
+
+        public override void OnInspectorGUI()
+        {
+            EditorGUILayout.HelpBox(k_Message, MessageType.Warning);
+
+            using (new EditorGUI.DisabledScope(true))
+            {
+                DrawDefaultInspector();
```

Comment:
In this special case this is OK.

Though, here is a few things to take into consideration.
- This only works if there is no specific editor for Projector. If there is a dedicated editor for it in built-in, display can differ from default one. (can be different placement, tooltips, edit mods etc).
- Simplest is to find if there is a [CustomEditor(typeof(Projector))] somewhere in Unity. If there is one, you will have differences. In this case, I recommend instead to create an Inspector for Projector using Editor.CreateEditor and display it here to be sure the display remain the same.

Here I think just a small comment to explain why we do not have to do this could help in case of maintenance needed on it.

Reply:
Thanks, I'll add a comment that there's no custom editor for Projector.

> For customization of message, if all in CoreRP, you can grab the render pipeline asset currently used and check its type I guess.
As far as I understand, existence of CoreRP package doesn't imply either URP or HDRP package are available, which is where those types are defined. Can I do this without hard-coding the typenames ("UniversalRenderPipelineAsset" and "HighDefinitionRenderPipelineAsset")?